### PR TITLE
[5.4] Create tracing on Eloquent Models

### PR DESCRIPTION
**Pull request by filipeaclima**
Similar to what Laravel has currently with timestamps ('created_at' /
'updated_at') on models, I propose the creation of a (very) basic user
action tracing system through 'created_by' (for the author) and
'updated_by' (for the last editor) attributes on models.

I've had this need on a project I'm working on, and thought that maybe
it could be useful to someone else.

TODO: add methods on Model class to retrieve the author instance and
last editor instance. I have to dig a bit on this as the user provider
depends on the provider that is configured. Just wanted to check
with you if this is valid or not before I do that.

### CODE REVIEW

File: src/Illuminate/Database/Eloquent/Model.php [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 namespace Illuminate\Database\Eloquent;
 
+use Auth;
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Model.php]: that class doesn't exist

File: src/Illuminate/Database/Eloquent/Model.php [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 namespace Illuminate\Database\Eloquent;
 
+use Illuminate\Support\Facades\Auth;
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Model.php]: We can't add this dependency.

**Comment by GrahamCampbell** [comment]
This probably should be implemented as an observer in your own app.

**Comment by filipeaclima** [comment]
On my app I'm doing it with observers as you suggested, I just used the Model class here as it seemed the logical place for a feature like this if it was bundled with Laravel. But if we can't have the dependency there (I now see why) maybe it doesn't make sense at all to have this included on Laravel.

**Comment by rs-sliske** [comment]
shameless plug, but i do have a package which provides an alternative approach to this. https://github.com/RMoorePHP/change-recorder

**Comment by taylorotwell** [comment]
No plans to add this to core.

**Pull request closed without merging**